diff --git a/Core/Src/main.c b/Core/Src/main.c
index 0c771ff..346accc 100644
--- a/vistron_vdr_400/Core/Src/main.c
+++ b/vistron_vdr_400/Core/Src/main.c
@@ -224,9 +224,24 @@ int main(void)
 
   /* USER CODE BEGIN SysInit */
 
-#ifdef FREE_RTOS
+  /* USER CODE END SysInit */
 
-  __HAL_RCC_GPIOE_CLK_ENABLE();
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+#ifndef FREE_RTOS
+  MX_RTC_Init();
+  MX_I2C2_Init();
+  MX_SPI2_Init();
+#endif
+  MX_TIM5_Init();
+#ifndef FREE_RTOS
+  MX_TIM6_Init();
+  MX_USART2_UART_Init();
+  MX_FSMC_Init();
+  /* USER CODE BEGIN 2 */
+#endif
+
+#ifdef FREE_RTOS
 
   TaskHandle_t led1Task;
   (void) xTaskCreate( LED1_Task, "LED1", LED_TASK_STACK_SIZE,
@@ -240,19 +255,6 @@ int main(void)
 
 #else //FREE_RTOS
 
-  /* USER CODE END SysInit */
-
-  /* Initialize all configured peripherals */
-  MX_GPIO_Init();
-  MX_RTC_Init();
-  MX_I2C2_Init();
-  MX_SPI2_Init();
-  MX_TIM5_Init();
-  MX_TIM6_Init();
-  MX_USART2_UART_Init();
-  MX_FSMC_Init();
-  /* USER CODE BEGIN 2 */
-
   ili9341_init();
 
   touch_init();
@@ -909,6 +911,8 @@ static void MX_SPI2_Init(void)
 
 }
 
+#endif //FREE_RTOS
+
 /**
   * @brief TIM5 Initialization Function
   * @param None
@@ -956,6 +960,8 @@ static void MX_TIM5_Init(void)
 
 }
 
+#ifndef FREE_RTOS
+
 /**
   * @brief TIM6 Initialization Function
   * @param None
@@ -1031,6 +1037,8 @@ static void MX_USART2_UART_Init(void)
 
 }
 
+#endif //FREE_RTOS
+
 /**
   * @brief GPIO Initialization Function
   * @param None
@@ -1148,6 +1156,8 @@ static void MX_GPIO_Init(void)
 
 }
 
+#ifndef FREE_RTOS
+
 /* FSMC initialization function */
 static void MX_FSMC_Init(void)
 {
