diff --git a/Core/Src/main.c b/Core/Src/main.c
index b6d8ec5..14899eb 100644
--- a/vistron_vdr_400/Core/Src/main.c
+++ b/vistron_vdr_400/Core/Src/main.c
@@ -26,6 +26,10 @@
 #include "../../User/func_8001ae8.h"
 #include <stdlib.h>
 
+#ifdef FREE_RTOS
+#include "FreeRTOS.h"
+#endif
+
 /* USER CODE END Includes */
 
 /* Private typedef -----------------------------------------------------------*/
@@ -83,6 +87,9 @@ extern int sub_800a9a8(void);
 /* 800c7e0 - todo */
 void sub_800c7e0(uint16_t a)
 {
+#ifdef FREE_RTOS
+   vTaskDelay(pdMS_TO_TICKS(a));
+#else
    TIM_HandleTypeDef htim;
 
    htim.Instance = TIM6;
@@ -100,6 +107,7 @@ void sub_800c7e0(uint16_t a)
    }
 
    HAL_TIM_Base_Stop_IT(&htim);
+#endif
 }
 
 #ifndef FREE_RTOS
